client/components/site-selector-modal/index.jsx

```diff
@@ -35,15 +37,25 @@ var SiteSelectorModal = React.createClass( {
 	},
 
 	getInitialState: function() {
-		return ( {
-			site: sitesList.getPrimary().jetpack ?
-				sitesList.get().filter( this.props.filter )[0] :
-				sitesList.getPrimary()
-		} );
+		const primarySite = sitesList.getPrimary();
+		let filteredSites = sitesList.get();
```

Comment:
Might change this to `getVisible()` to get rid of the ugly filter combining thing below.